File: frontend/naomie1/src/app/models/project.ts

export class Project {

  public scenarios : string[];

  constructor(public name: string, public key: string, public userName: string,
    public creationDate: string, public summary:string,
    scenarioConcat : string){

      this.scenarios = scenarioConcat.split(" ")
    };
}

export class ProjectForm {
  constructor(public name: string, public userName: string, public creationDate: string, public summary:string){};
}

File: backend/naomie/src/main/scala/com/github/GuillaumeReinisch/Naomie/models/project.scala
package com.github.GuillaumeReinisch.Naomie.models

import com.google.cloud.datastore.Entity
import org.slf4j.LoggerFactory

/**
  * Created by lcts on 12/03/17.
  */
case class Project( name : String,key : String, userName: String, creationDate: String, summary: String, scenarios : List[String] = List.empty[String],  graphics : List[String] = List.empty[String]) {

}


object Project {
  val logger    =  LoggerFactory.getLogger(getClass)
  def apply( entity : Entity) : Project = {

    Project(entity.getString("name"),
      entity.getString("name"),
      entity.getString("userName"),
      entity.getString("creationDate"),
      entity.getString("summary"),
      entity.getString("scenarios").split(" ").toList,
      entity.getString("graphics").split(" ").toList
    )
  }
}
